.rubocop.yml

```diff
@@ -0,0 +1,520 @@
+---
+require:
+- rubocop-performance
+- rubocop-rspec
+AllCops:
+  DisplayCopNames: true
+  TargetRubyVersion: '2.5'
+  SuggestExtensions: false
+  Include:
+  - "**/*.rb"
+  Exclude:
+  - bin/*
+  - ".vendor/**/*"
+  - "**/Gemfile"
+  - "**/Rakefile"
+  - pkg/**/*
+  - spec/fixtures/**/*
+  - vendor/**/*
+  - "**/Puppetfile"
+  - "**/Vagrantfile"
+  - "**/Guardfile"
+Layout/LineLength:
```

Comment:
I'm not a fan of this HUGE config. Any thoughts about reusing it like we did with our Puppet modules?

For reference: https://github.com/voxpupuli/modulesync_config/blob/f14a18d2ae681a4c973b3e898ab16e384046ca45/moduleroot/.rubocop.yml.erb#L8-L9

This means we can put all the rules in puppet-lint.gem and use it in every plugin. Note that you can still override everything in gems, you just set a baseline.

Reply:
I like this idea a lot. If i'm honest, I didn't even realise it was a possibility.

I guess we could sync the merge of this PR with a new release of puppet-lint.